src/openrct2/ride/RideRatings.cpp

```diff
@@ -160,20 +160,20 @@ static void ride_ratings_update_state(RideRatingUpdateState& state)
  */
 static void ride_ratings_update_state_0(RideRatingUpdateState& state)
 {
-    ride_id_t currentRide = state.CurrentRide;
+    auto currentRide = state.CurrentRide;
 
-    currentRide = static_cast<ride_id_t>(EnumValue(currentRide) + 1);
-    if (currentRide >= static_cast<ride_id_t>(MAX_RIDES))
+    auto nextRide = RideId::FromUnderlying(currentRide.ToUnderlying() + 1);
```

Comment:
Perhaps it would be a good idea to implement `operator++` for TIdentifier?

Reply:
I don't think this instance warrants that, I rather have this code cleaned up to use the RideManager and its iterators, they already implement this. Definitely for a future PR